src/framework/mpas_dmpar.F

```diff
@@ -6167,6 +6208,218 @@ subroutine mpas_dmpar_exch_halo_field5d_real(field, haloLayersIn)!{{{
+                  do  i = 1, exchListPtr % nList
+                    do j = 1, fieldCursor % dimSizes(1)
+                      commListPtr % rbuffer((exchListPtr % srcList(i)-1) * fieldCursor % dimSizes(1) + j + bufferOffset) = fieldCursor % array(j, exchListPtr % destList(i))
+                      ! update halo cell
+                      fieldCursor % array(j, exchListPtr % destList(i)) = 0.0_RKIND
+                      nAdded = nAdded + 1
+                    end do
+                  end do
+                end if
+
+                exchListPtr => exchListPtr % next
+              end do
+
+              fieldCursor => fieldCursor % next
+            end do
+            bufferOffset = bufferOffset + nAdded
+          end do
+
+          call MPI_Isend(commListPtr % rbuffer, commListPtr % nList, MPI_REALKIND, commListPtr % procID, dminfo % my_proc_id, dminfo % comm, commListPtr % reqID, mpi_ierr)
+          commListPtr => commListPtr % next
+        end do
+#endif
+
+        ! Handle local copy. If MPI is off, then only local copies are performed.
+        fieldCursor => field
+        do while(associated(fieldCursor))
+          do iHalo = 1, nHaloLayers
+            exchListPtr => fieldCursor % copyList % halos(haloLayers(iHalo)) % exchList
+
+            do while(associated(exchListPtr))
+              fieldCursor2 => field
+              do while(associated(fieldCursor2))
+                if(exchListPtr % endPointID == fieldCursor2 % block % localBlockID) then
+                  do i = 1, exchListPtr % nList
+                    !fieldCursor2 % array(:, exchListPtr % destList(i)) = fieldCursor % array(:, exchListPtr % srcList(i))
+                    fieldCursor % array(:, exchListPtr % srcList(i)) = fieldCursor % array(:, exchListPtr % srcList(i)) + fieldCursor2 % array(:, exchListPtr % destList(i))
+                    fieldCursor2 % array(:, exchListPtr % destList(i)) = 0.0_RKIND
+                  end do
+                end if
+
+                fieldCursor2 => fieldCursor2 % next
+              end do
+
+              exchListPtr => exchListPtr % next
+            end do
+          end do
+
+          fieldCursor => fieldCursor % next
+        end do
+
+#ifdef _MPI
+
+        ! Wait for mpi_irecv to finish, and unpack data from buffer
+        commListPtr => sendList
+        do while(associated(commListPtr))
+          call MPI_Wait(commListPtr % reqID, MPI_STATUS_IGNORE, mpi_ierr)
+          bufferOffset = 0
+          do iHalo = 1, nHaloLayers
+            nAdded = 0
+            fieldCursor => field
+            do while(associated(fieldCursor))
+              exchListPtr => fieldCursor % sendList % halos(haloLayers(iHalo)) % exchList
+              do while(associated(exchListPtr))
+                if(exchListPtr % endPointID == commListPtr % procID) then
+                  do i = 1, exchListPtr % nList
+                    if (present(interiorCells)) then
```

Comment:
Beyond debugging and testing, is there a strong argument for including this code (when `interiorCells` is present)?

Reply:
There is not, it was only for debugging. I will remove it from this routine (as I will remove it from the corresponding `mpas_dmpar_exch_halo_field2d_real`